### ISSUE TITLE
panic in HTTP ingester if no listeners are specified

### ISSUE THREAD

**Issue by traetox** [gravwell/gravwell#341]
Here is a `gravwell_http_ingester.conf` config:
```
Global]
Ingester-UUID="941af247-e3af-494b-8ccf-398f0636a81c"
Ingest-Secret = IngestSecrets
Connection-Timeout = 0
Insecure-Skip-TLS-Verify=true
Pipe-Backend-Target=/opt/gravwell/comms/pipe #a named pipe connection, this should be used when ingester is on the same machine as a backend
Log-Level=INFO #options are OFF INFO WARN ERROR
Ingest-Cache-Path=/opt/gravwell/cache/http_ingester.cache
#Max-Ingest-Cache=1024 #Number of MB to store, localcache will only store 1GB before stopping.  This is a safety net
Bind=":8080"
Max-Body=4096000 #about 4MB
Log-File=/opt/gravwell/log/http_ingester.log #optional log file
Health-Check-URL="/health/check"

# Example that creates a listener that is API compatible with the Amazon Kinesis Delivery Stream
#[Kinesis-Delivery-Stream-Listener "testing"]
[foobar "testing"]
	URL="/kinesis/stream/foobar"
	TokenName="Gravwell" #set this to your token name
	TokenValue="on6O6ekLYixvlgDXdV7dprvapHIzxcQK" #set the access control token
	Tag-Name=KDSStuff
```

Startup the HTTP ingester and you get this crash:
```
panic: warnings.Collector already done

goroutine 1 [running]:
gopkg.in/warnings%2ev0.(*Collector).Collect(0xc000035110, {0x103e1c0, 0xc000361940})
	/home/kris/mygo/pkg/mod/gopkg.in/warnings.v0@v0.1.2/warnings.go:141 +0x18a
github.com/gravwell/gcfg.readIntoPass(0x7a4526, {0xd20f60, 0xc0003024e0}, 0xc000236f80, 0xe31500, {0xc0000d7400, 0x3db, 0x400}, 0x8)
	/home/kris/mygo/pkg/mod/github.com/gravwell/gcfg@v1.2.9-0.20210818172109-3d05a45a2665/read.go:132 +0x775
github.com/gravwell/gcfg.readInto({0xd20f60, 0xc0003024e0}, 0x0, 0x1, {0xc0000d7400, 0x3db, 0x400})
	/home/kris/mygo/pkg/mod/github.com/gravwell/gcfg@v1.2.9-0.20210818172109-3d05a45a2665/read.go:199 +0xae
github.com/gravwell/gcfg.ReadInto({0xd20f60, 0xc0003024e0}, {0x103fc80, 0xc0003001e0})
	/home/kris/mygo/pkg/mod/github.com/gravwell/gcfg@v1.2.9-0.20210818172109-3d05a45a2665/read.go:219 +0xb9
github.com/gravwell/gcfg.ReadStringInto(...)
	/home/kris/mygo/pkg/mod/github.com/gravwell/gcfg@v1.2.9-0.20210818172109-3d05a45a2665/read.go:226
github.com/gravwell/gravwell/v3/ingest/config.LoadConfigBytes({0xd20f60, 0xc0003024e0}, {0xc000366000, 0xc00000e7b8, 0x0})
	/home/kris/githubwork/gravwell/ingest/config/loader.go:81 +0xb8
github.com/gravwell/gravwell/v3/ingest/config.LoadConfigFile({0xd20f60, 0xc0003024e0}, {0xea6132, 0x44f994})
	/home/kris/githubwork/gravwell/ingest/config/loader.go:71 +0x247
main.GetConfig({0xea6132, 0x2d})
	/home/kris/githubwork/gravwell/ingesters/HttpIngester/config.go:76 +0x6c
main.main()
	/home/kris/githubwork/gravwell/ingesters/HttpIngester/main.go:88 +0x65
```

**Comment by floren** [comment]
This is a bug in our gcfg.

**Comment by floren** [comment]
Fixed by #342
